src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -485,11 +489,13 @@ private async Task ExecuteTestsInSourceAsync(IEnumerable<TestCase> tests, IRunCo
             // Queue the non parallel set
             if (nonParallelizableTestSet != null)
             {
+                // What to do?
                 await ExecuteTestsWithTestRunnerAsync(nonParallelizableTestSet, frameworkHandle, source, sourceLevelParameters, testRunner, usesAppDomains);
             }
         }
         else
         {
+            // What to do?
```

Comment:
Here and few lines above we run the tests when parallelization is not enabled or test is marked with do not parallelize, this blocks the cancellation until the test finishes (and fails the non-parallel tests), and advice here how elegantly solve this? (it won't block if we have parallelism with 1 worker, because that takes the other parallel path.

Reply:
@nohwnd How does it block cancellation? `ExecuteTestsWithTestRunnerAsync` calls `ThrowIfCancellationRequested`

Reply:
Yes but only before it starts the run. if we cancel while the test is running we don't have a way to abandon it to background and continue to teardown code.

Reply:
You can still do the same Task.WhenAny trick right?